examples/mujoco/td3/train_td3.py

```diff
@@ -0,0 +1,202 @@
+"""A training script of TD3 on OpenAI Gym Mujoco environments.
+
+This script follows the settings of http://arxiv.org/abs/1802.09477 as much
+as possible.
+"""
+from __future__ import print_function
+from __future__ import unicode_literals
+from __future__ import division
+from __future__ import absolute_import
+from future import standard_library
+standard_library.install_aliases()  # NOQA
+
+import argparse
+import logging
+import os
+import sys
+
+import chainer
+from chainer import functions as F
+from chainer import links as L
+from chainer import optimizers
+import gym
+import gym.wrappers
+import numpy as np
+
+import chainerrl
+from chainerrl import experiments
+from chainerrl import explorers
+from chainerrl import misc
+from chainerrl import replay_buffer
+
+
+def concat_obs_and_action(obs, action):
+    """Concat observation and action to feed the critic."""
+    return F.concat((obs, action), axis=-1)
+
+
+def main():
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--outdir', type=str, default='results',
+                        help='Directory path to save output files.'
+                             ' If it does not exist, it will be created.')
+    parser.add_argument('--env', type=str, default='Hopper-v2',
+                        help='OpenAI Gym MuJoCo env to perform algorithm on.')
+    parser.add_argument('--seed', type=int, default=0,
+                        help='Random seed [0, 2 ** 32)')
+    parser.add_argument('--gpu', type=int, default=0,
+                        help='GPU to use, set to -1 if no GPU.')
+    parser.add_argument('--load', type=str, default='',
+                        help='Directory to load agent from.')
+    parser.add_argument('--steps', type=int, default=10 ** 6,
+                        help='Total number of timesteps to train the agent.')
+    parser.add_argument('--eval-n-runs', type=int, default=10,
+                        help='Number of episodes run for each evaluation.')
+    parser.add_argument('--eval-interval', type=int, default=5000,
+                        help='Interval in timesteps between evaluations.')
+    parser.add_argument('--replay-start-size', type=int, default=10000,
```

Comment:
Where did you get this number? I didn't see it in the paper. I know they do pure exploration for 10K frames, but I didn't see that they don't do any updates for 10K frames.

Reply:
They actually updated the model N times after every episode where N is the length of that episode. https://github.com/sfujim/TD3/blob/master/main.py#L94 So, this is different from what they did. I don't think this difference significantly affects, but I'll add explanation on this.